ttnn/cpp/ttnn/operations/experimental/cnn/convert_to_chw/device/convert_to_chw_program_factory.cpp

```diff
@@ -47,16 +56,22 @@ operation::ProgramWithCallbacks multi_core_convert_to_chw(
     const uint32_t intermediary_tile_size = tt::tt_metal::detail::TileSize(intermediary_format);
 
     const uint32_t cb_in_id = tt::CB::c_in0;
-    const auto cb_in =
-        create_circular_buffer(cb_in_id, total_tiles_per_core, input_tile_size, input_format, a.buffer());
+    const uint32_t cb_in_total_size = total_tiles_per_core * input_tile_size;
+    const uint32_t cb_in_page_size = input_tile_size;
+    const auto cb_in = create_circular_buffer(cb_in_id, cb_in_total_size, cb_in_page_size, input_format, a.buffer());
 
     const uint32_t cb_out_id = tt::CB::c_out0;
+    const uint32_t element_size = tt::datum_size(input_format);
+    const uint32_t cb_out_total_size = C * HW * element_size / input_cores.size();
+    const uint32_t cb_out_page_size = HW / input_cores.size();
```

Comment:
Is ROW_MAJOR guaranteed? If not, definitely don't use `element_size`. 
Also, `cb_out_page_size` looks wrong to me. There's no mention of size in bytes anywhere.

Reply:
Did you push your changes? I don't see it

Reply:
Apparently I forgot to do that, pushed now!